frontend/src/lib/components/PropertyFilters/components/TaxonomicPropertyFilter.tsx

```diff
@@ -205,7 +205,7 @@ export function TaxonomicPropertyFilter({
                                 onClick={() => (dropdownOpen ? closeDropdown() : openDropdown())}
                             >
                                 {filter?.type === 'cohort' ? (
-                                    selectedCohortName || `Cohort #${filter?.value}`
+                                    filter.cohort_name || `Cohort #${filter?.value}`
```

Comment:
this fixes the issue described in the thread with loading a flag that's already been saved.